mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -1092,7 +1092,7 @@ static LogicalResult addInitOperandsToLoopNest(
   for (auto [outerLoop, innerLoop] :
        llvm::zip_equal(loops.drop_back(), loops.drop_front())) {
     // Again assume that all the outer loops are scf.for operations.
-    auto outerForLoop = cast<scf::ForOp>(outerLoop);
+    auto outerForLoop = cast<scf::ForOp>(outerLoop.getOperation());
```

Comment:
This looks not related to the PR itself, we can consider reverting it.

Reply:
I was getting a warning/error without this for whatever reason. Adding this fixed it. I didnt look further as to why cause this seemed reasonable enough thing to do.